contracts/test/unit/oracle/UniswapV3OracleWrapper.t.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {IMockUniswapV3Pool} from "../../../mock/IMockUniswapV3Pool.sol";
+import {IOracle, Decimal} from "../../../oracle/IOracle.sol";
+import {ICore} from "../../../core/ICore.sol";
+import {IUniswapWrapper} from "../../../oracle/uniswap/IUniswapWrapper.sol";
+
+import {MockERC20} from "../../../mock/MockERC20.sol";
+import {UniswapV3OracleWrapper} from "../../../oracle/uniswap/UniswapV3OracleWrapper.sol";
+import {FeiTestAddresses, getAddresses, getCore} from "../../utils/fixtures/FeiFixture.sol";
+
+import {DSTest} from "../../utils/DSTest.sol";
+import {StdLib} from "../../utils/StdLib.sol";
+import {Vm} from "../../utils/Vm.sol";
+
+// Note: Where deployCode() is used, it's a workaround to deploy a contract necessarily compiled with a 
+// different Solidity version (Uniswap contracts written in different version, and rely on prior features)
+
+contract UniswapV3OracleTest is DSTest, StdLib {
+  IMockUniswapV3Pool private mockUniswapPool;
+  address private uniswapMathWrapper;
+  ICore core;
+
+  uint32 private twapPeriod = 61;
+  uint256 private precision = 10**18;
+
+  UniswapV3OracleWrapper private oracle;  
+  FeiTestAddresses addresses = getAddresses();
+  Vm public constant vm = Vm(HEVM_ADDRESS);
+
+  MockERC20 private tokenA = new MockERC20();
+  MockERC20 private tokenB = new MockERC20();
+  
+  
+  function setUp() public {
+    core = getCore();
+    mockUniswapPool = IMockUniswapV3Pool(deployCode("./out/MockUniV3Pool.sol/MockUniV3Pool.json"));
```

Comment:
why this weird syntax ?

Reply:
The `Uniswap` contracts are compiled with a different version of Solidity to what we use. This test file is compiled with Solidity > 0.8, whereas the Uniswap contracts use < 0.8. 

That means you can't import the Uniswap related contracts into this test Solidity file and compile. Forge has a nice way around this where you can pull the pre-compiled artifact and deploy it with `deployCode()`